modules/calc/src/main/java/com/opengamma/strata/calc/runner/TaskColumn.java

```diff
@@ -0,0 +1,228 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.calc.runner;
+
+import java.util.Set;
+
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.light.LightMetaBean;
+
+import com.opengamma.strata.basics.CalculationTarget;
+import com.opengamma.strata.calc.Column;
+import com.opengamma.strata.calc.config.MarketDataRules;
+import com.opengamma.strata.calc.config.Measure;
+import com.opengamma.strata.calc.config.ReportingCurrency;
+import com.opengamma.strata.calc.marketdata.mapping.MarketDataMappings;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.Messages;
+
+/**
+ * A single column used to build the calculation tasks.
+ * <p>
+ * Each {@link CalculationTask} calculates a grid of results.
+ * This class capture details of each column.
+ */
+@BeanDefinition(style = "light")
+public final class TaskColumn implements ImmutableBean {
+
+  /**
+   * The column index of the cell in the results grid.
+   */
+  @PropertyDefinition(validate = "ArgChecker.notNegative")
+  private final int columnIndex;
+  /**
+   * The measure to be calculated.
+   * <p>
+   * This defines the calculation being performed, such as 'PresentValue' or 'ParRate'.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final Measure measure;
+  /**
+   * The calculation parameters, used to control the how the calculation is performed.
+   * <p>
+   * The parameters specified here must be specific to the target and measure.
+   * The reporting currency parameter will not be present.
```

Comment:
This comment makes me question whether reporting currency should be a parameter at all. It is obviously already receiving special treatment. It would seem clearer if reporting currency were fully separated and parameters only applied to targets and measures.